ADAL/src/request/ADWebRequest.h

```diff
@@ -54,6 +54,7 @@ typedef void (^ADWebResponseCallback)(NSMutableDictionary *);
 }
 
 @property (strong, readonly, nonatomic) NSURL               *URL;
+@property (copy, nonatomic)             NSMutableDictionary *headers;
```

Comment:
Correct, we do not want to provide direct access to the dictionary itself. You could add a -addToHeadersFromDictionary: method if you wish, but we shouldn't be providing direct access to a mutable dictionary property like this as it tends to cause abuse.